status-payloads-spec.md

```diff
@@ -149,10 +141,91 @@ All incoming messages can be matched against a chat. Below you can find a table
+
+### SyncInstallation
+
+`SyncInstallation` messages are used to synchronize in a best-effort way all the paired installations. It is identified by a transit tag of `p1` 
+
+#### Payload
+
+Payload is a struct (a compound data type) with the following fields (order is important):
+
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1| contacts | `[struct { name string last-updated int device-info struct {id string fcm-token string } pending? bool}` | An array of contacts |
+| 2 | account | `struct {name string photo-path string last-updated int}` | Information about your own account |
+| 3 | chat | `struct {:public? bool :chat-id string}` | A description of a public chat opened by the client |
+
+### PairInstallation
+
+`PairInstallation` messages are used to propagate informations about a device to its paired devices. It is identified by a transit tag of `p2` 
+
+#### Payload
+
+Payload is a struct (a compound data type) with the following fields (order is important):
+
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1| installation-id | `string` | A randomly generated id that identifies this device |
+| 2 | device-type | `string` | The OS of the device `ios`,`android` or `desktop` |
+| 3 | name | `string` | The self-assigned name of the device |
+| 4 | fcm-token | `string` | The FCM Token used by mobile devices for push notifications |
+
+### GroupMembershipUpdate
+
+`GroupMembershipUpdate` is a message used to propagate information about group membership changes in a group chat.. It is identified by a transit tag of `g5`.
+The details are in the  [Group chats specs](status-group-chats-spec.md)
+
+#### Payload
+
+Payload is a struct (a compound data type) with the following fields (order is important):
+
+
+| Field | Name | Type | Description |
+| ----- | ---- | ---- | ---- |
+| 1| chat-id | `string` | The chat id of the chat where the change is to take place |
+| 2 | membership-updates | See details | A list of events that describe the membership changes |
+| 3 | message | `Transit message` | An optional message, described in [Message](#message) |
+
 ## Upgradability
 
-The current protocol format is hardly upgradable without breaking backward compatibility. Because Transit is used in this particular way described above, the only reliable option is to append a new field to the Transit record definition. It will be simply ignored by the old clients.
-<!-- Not sure I agree with this statement, seems very arbitrary, appending to an array is just as upgradable as adding an entry in a map, just less convenient, I would remove the qualitative statement, and just describe how to upgrade -->
+There are two ways to upgrade the protocol without breaking compatibility:
+
+- Map fields can be enriched with a new key, which will be ignored by old clients.
```

Comment:
what are map fields?

Reply:
fields that are hashmaps/associative arrays/structs i.e {"something-like-this": "a"} maybe best to call it a struct for consistency?

Reply:
I do like consistency, but if the naming is specifically appropriate then we can use it, as there are some subtle differences to those.